doc/modules/decomposition.rst

```diff
@@ -278,6 +257,88 @@ factorization, while larger values shrink many coefficients to zero.
      R. Jenatton, G. Obozinski, F. Bach, 2009
 
 
+.. _kernel_PCA:
+
+Kernel Principal Component Analysis (kPCA)
+==========================================
+
+Exact Kernel PCA
+----------------
+
+:class:`KernelPCA` is an extension of PCA which achieves non-linear
+dimensionality reduction through the use of kernels (see :ref:`metrics`). It
+has many applications including denoising, compression and structured
+prediction (kernel dependency estimation). :class:`KernelPCA` supports both
+``transform`` and ``inverse_transform``.
+
+.. figure:: ../auto_examples/decomposition/images/sphx_glr_plot_kernel_pca_001.png
+    :target: ../auto_examples/decomposition/plot_kernel_pca.html
+    :align: center
+    :scale: 75%
+
+.. topic:: Examples:
+
+    * :ref:`sphx_glr_auto_examples_decomposition_plot_kernel_pca.py`
+
+.. _Randomized_kPCA:
+
+Kernel PCA using randomized SVD
+-------------------------------
+
+In :ref:`kernel_PCA`, the number of components found is equal to the number of
+samples. Many real-world datasets have large number of samples ! In these cases
+finding *all* the components with a full kPCA is a waste of computation time,
+as data is mostly described by the first few components
+(``n_components<=100``).
+
+The optional parameter ``eigen_solver='randomized'`` can be used to
+*drastically* reduce the computation time when the number of requested
```

Comment:
```suggestion
reduce the computation time when the number of requested
```
benchmarks suggest it's not that drastic I would say, i.e. not multiple orders of magnitude faster

Reply:
I would need confirmation from an expert, but it should be O(`n_components`/`n_samples`) faster. I call this "drastically" when n comps is < 100 and n_samples is > 10000... But it is a matter of language

Reply:
Use "significantly"? I don't think we should bike shed on this...

For info, this is an entry of 0.23 

- |Fix| |Efficiency| :class:`linear_model.ARDRegression` is more stable and
  much faster when `n_samples > n_features`

where "much faster" actually means "you couldn't fit more than 1k samples before but now this scales up to more than 500k samples" in 1 sec

Reply:
sure, I'll use significantly. I learned [a new expression](https://www.urbandictionary.com/define.php?term=bikeshed), thanks !